### DISCUSSION TITLE
any plan to support client concurrent write data to corresponding server?

### DISCUSSION THREAD

**Discussion by vklonghml** [minio/minio#18804] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
currently, mulitipart write all data to a one server, then the server does the EC coding, and send the data to other server in the Set, there is a bottleneck between client and the server who received the data.

**Describe the solution you'd like**
client does the EC coding, and send the data to the corresponding server concurrently

**Comment by klauspost** [comment, marked as answer]
We have been investigating this: https://github.com/minio/minio/pull/15781

However the complexity has not yet been met by a clear enough business need. Feel free to reach out on `hello@min.io` if you have a business need for it.
